docs/building.md

```diff
@@ -18,17 +18,106 @@ If you come across a disfunctional setup, please report it by open an issue or p
+   | PARTHENON\_DISABLE\_EXAMPLES | OFF      | Option | Toggle building of examples, if regression tests are on, drivers needed by the tests will still be built |
    |   ENABLE\_INTEGRATION\_TESTS | ${BUILD\_TESTING} | Option | Enable integration tests |
    |    ENABLE\_REGRESSION\_TESTS | ${BUILD\_TESTING} | Option | Enable regression tests |
    |  REGRESSION\_GOLD\_STANDARD\_VER | #     | Int    | Version of current gold standard file used in regression tests. Default is set to latest version matching the source. |
    | REGRESSION\_GOLD\_STANDARD\_HASH | SHA512=... | String | Hash value of gold standard file to be downloaded. Used to ensure that the download is not corrupted. |
    | REGRESSION\_GOLD\_STANDARD\_SYNC | ON    | Option | Create `gold_standard` target to download gold standard files |
    |          ENABLE\_UNIT\_TESTS | ${BUILD\_TESTING} | Option | Enable unit tests |
    |               CODE\_COVERAGE | OFF      | Option | Builds with code coverage flags |
+   |       CMAKE\_INSTALL\_PREFIX | machine specific | String | Optional path for library installation |
+   |  PARTHENON\_EXTERNAL\_KOKKOS | ON/OFF    | Option | Whether to use external Kokkos installation |
+   |          BUILD\_SHARED\_LIBS | OFF       | Option | If installing Parthenon, whether to build as shared rather than static |
 
 ### NB: CMake options prefixed with *PARTHENON\_* modify behavior.
 
+## Installing Parthenon
+
+An alternative to building Parthenon alongside a custom app (as in the examples)
+is to first build Parthenon separately as a library and then link to it
+when building the app. Parthenon can be built as either a static (default) or a shared library.
+
+To build Parthenon as a library, provide a `CMAKE_INSTALL_PREFIX` path
+to the desired install location to the Parthenon cmake call. To build a shared rather
+than a static library, also set `BUILD_SHARED_LIBS=ON`. Then build and install
+(note that `--build` and `--install` require CMake 3.15 or greater).
+
+### Building as a static library
+
+```bash
+cmake -DCMAKE_INSTALL_PREFIX="$your_install_dir" $parthenon_source_dir
+cmake --build . --parallel
+cmake --install .
+```
+
+### Building as a shared library
+
+```bash
+cmake -DCMAKE_INSTALL_PREFIX="$your_install_dir" -DBUILD_SHARED_LIBS=ON $parthenon_source_dir
+cmake --build . --parallel
+cmake --install .
+```
+
+When building Parthenon, Kokkos will also be built from source if it exists in
+`parthenon/external` or at a provided `Kokkos_ROOT` by default. If installing
+Parthenon, this will also install Kokkos in the same directory. If
+`PARTHENON_EXTERNAL_KOKKOS=ON` is provided or no Kokkos/CMakeLists.txt is found,
+the build system will attempt to find a Kokkos installation in the current PATH.
+
+A cmake target, `lib/cmake/parthenon/parthenonConfig.cmake` is created during
```

Comment:
Ok, I see you added a description here, I would still recommend updating the option description.

Reply:
Is changing the option description in the documentation to
```
   |                 Kokkos\_ROOT | unset    | String | Path to a Kokkos source directory (containing CMakeLists.txt) |
   |  PARTHENON\_EXTERNAL\_KOKKOS | ON/OFF   | Option | Whether to attempt to link to an extneral Kokkos installation |
```
and the `CMakeLists.txt` option label to
```
# Default to an external Kokkos package if the submodule is not populated        
if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos/CMakeLists.txt" AND NOT EXISTS "${Kokkos_ROOT}/CMakeLists.txt")
    option(PARTHENON_EXTERNAL_KOKKOS "Attempt to link to an external Kokkos installation" ON)
else()                                                                           
    option(PARTHENON_EXTERNAL_KOKKOS "Attempt to link to an external Kokkos installation" OFF)
endif()
```
what you had in mind? Sorry if I misunderstood.

Reply:
In both cases I would say something along the lines of:

"If ON attempt to link to an external Kokkos library, if OFF build and package Kokkos from source with parthenon."

Reply:
That's better, thanks